src/ui/public/saved_objects/saved_object.js

```diff
@@ -0,0 +1,35 @@
+import _ from 'lodash';
+
+export class SavedObject {
+  constructor(client, { id, type, version, attributes } = {}) {
+    this._client = client;
+    this.id = id;
+    this.type = type;
+    this._attributes = attributes || {};
+    this._version = version;
+  }
+
+  get(key) {
+    return _.get(this._attributes, key);
+  }
+
+  set(key, value) {
+    return _.set(this._attributes, key, value);
+  }
+
+  has(key) {
+    return _.has(this._attributes, key);
+  }
+
+  save() {
+    if (this.id) {
+      return this._client.update(this.type, this.id, this._attributes);
```

Comment:
In a discussion last week with @epixa, we decided against allowing creation with an explicit `id`. We will be migrating all of the existing ids to UUIDs and keeping a `legacyId` field to maintain support for legacy share URLs similar to a slug without committing to the slug feature at this time.